### DISCUSSION TITLE
get depth-map

### DISCUSSION THREAD

**Discussion by wiesehahn** [maplibre/maplibre-gl-js#1777] [Q&A]
I wonder if it is possible to get a depth-map via maplibre? Meaning the distance from the current camera position to each Pixel in the map. 
For untilted views from top this would basically be the distance to the ground plane and is only dependend on the zoom level for 2D maps and additionally on the terrain elevation for 3D maps. It gets more complicated for tilted maps with 3D terrain.

Applications I have in mind are e.g. to apply filters on the mapview based on distance. (Blur distant areas, desaturate further mountain ranges, ...).

![Depth-map-of-photo-in-Figure-5](https://user-images.githubusercontent.com/41429613/197890374-90fd03a4-b0b6-412d-89c5-d95aa9499ade.png)

**Comment by HarelM** [comment]
I'm not sure this is possible, but there is a sky PR which sounds similar to what you would like to achieve. I would consider looking into if first and see if it helps with this concept:
https://github.com/maplibre/maplibre-gl-js/pull/1713

**Comment by wiesehahn** [comment]
https://github.com/maplibre/maplibre-style-spec/issues/161 seems somehow related
